backend/api/src/infrastructure/simple_storage/mod.rs

```
@@ -0,0 +1,112 @@
+use std::{
+	collections::hash_map::DefaultHasher,
+	hash::{Hash, Hasher},
+	str::FromStr,
+};
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use rusoto_core::Region;
+use rusoto_s3::{PutObjectRequest, S3Client, StreamingBody, S3};
+use serde::Deserialize;
+use url::Url;
+
+use crate::domain::{ImageStoreService, ImageStoreServiceError};
+
+#[derive(Deserialize)]
+pub struct Config {
+	images_bucket_name: String,
+	bucket_region: String,
+}
+
+#[derive(Clone)]
+pub struct Client {
+	s3_client: S3Client,
+	images_bucket_name: String,
+}
+
+impl Client {
+	pub async fn new(config: &Config) -> Self {
+		let s3_client = S3Client::new(
+			Region::from_str(config.bucket_region.as_str())
+				.unwrap_or_else(|_| panic!("Invalid AWS region {}", config.bucket_region)),
```

I would rather rename the function to `build` and return a.`Result` to let the caller panic if needed. Here we do not control where we are called from.

---

Agreed